pkg/controller/common/association/resources.go

```diff
@@ -0,0 +1,73 @@
+package association
+
+import (
+	"context"
+
+	commonv1 "github.com/elastic/cloud-on-k8s/pkg/apis/common/v1"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/tracing"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/user"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
+	"go.elastic.co/apm"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// CreatedBy returns true if an object has been created by an other one
+type CreatedBy func(created, creator metav1.Object) bool
```

Comment:
It sounds like the APMServer would created a Secret for example, which is not exactly the case.
Maybe:

```go
// ServesAssociation returns true if the given resource has been created to serve the association of associatedResource
type ServesAssociation func(resource metav1.Object, associatedResource metav1.Object) bool
```